### ISSUE TITLE
Eliminated cast regression

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#8764]
The following should trace 0 on static platforms, but the manual cast to Float has been optimized out:

```haxe
class Main {

    static function foo( ?e : Float ) : Null<Single> {
		return (e:Float);
    }
    static function main() {
		trace( foo() );		
    }
}
```

**Comment by Simn** [comment]
This fails on the JVM target even with the cast.

**Comment by RealyUniqueName** [comment]
Merged into 4.0_bugfix
